dashboard/app/models/contact_rollups_final.rb

```diff
@@ -15,4 +15,27 @@
 
 class ContactRollupsFinal < ApplicationRecord
   self.table_name = 'contact_rollups_final'
+
+  def self.truncate_and_overwrite
+    truncate
+    overwrite_from_processed_table
+  end
+
+  def self.truncate
+    truncate_sql = <<~SQL
+      TRUNCATE contact_rollups_final;
+    SQL
+
+    ActiveRecord::Base.connection.exec_query(truncate_sql)
+  end
+
+  def self.overwrite_from_processed_table
```

Comment:
This function doesn't overwrite existing content, it just inserts new content. Maybe changing function name to match what it actually does.